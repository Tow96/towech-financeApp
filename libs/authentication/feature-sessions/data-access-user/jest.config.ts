/* eslint-disable */
export default {
  displayName: 'authentication-feature-sessions-data-access-user',
  preset: '../../../../jest.preset.js',
  testEnvironment: 'node',
  transform: {
    '^.+\\.[tj]s$': ['ts-jest', { tsconfig: '<rootDir>/tsconfig.spec.json' }],
  },
  moduleFileExtensions: ['ts', 'js', 'html'],
  coverageDirectory: '../../../../coverage/libs/authentication/feature-sessions/data-access-user',
  coverageReporters: ['html', 'text', 'text-summary', 'cobertura', 'lcov', 'json'],
  collectCoverageFrom: ['**/*.ts', '!**/*.module.ts', '!**/index.ts', '!**/jest.config.ts'],
};
