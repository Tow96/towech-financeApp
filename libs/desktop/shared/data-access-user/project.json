{
  "name": "desktop-shared-data-access-user",
  "$schema": "../../../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/desktop/shared/data-access-user/src",
  "prefix": "finance",
  "tags": [],
  "projectType": "library",
  "targets": {
    "test": {
      "executor": "@nx/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/{projectRoot}"],
      "options": {
        "jestConfig": "libs/desktop/shared/data-access-user/jest.config.ts",
        "passWithNoTests": true
      },
      "configurations": {
        "ci": {
          "ci": true,
          "codeCoverage": true
        }
      }
    },
    "lint": {
      "executor": "@nx/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": [
          "libs/desktop/shared/data-access-user/**/*.ts",
          "libs/desktop/shared/data-access-user/**/*.html"
        ]
      }
    }
  }
}
