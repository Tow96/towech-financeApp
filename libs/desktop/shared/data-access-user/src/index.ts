export * from './lib/user.service';
export * from './lib/types';
