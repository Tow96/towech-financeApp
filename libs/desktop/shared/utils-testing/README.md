# desktop-shared-utils-testing

This library was generated with [Nx](https://nx.dev).


## Running unit tests

Run `nx test desktop-shared-utils-testing` to execute the unit tests.

