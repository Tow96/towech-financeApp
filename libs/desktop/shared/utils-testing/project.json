{
  "name": "desktop-shared-utils-testing",
  "$schema": "../../../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/desktop/shared/utils-testing/src",
  "prefix": "finance",
  "tags": [],
  "projectType": "library",
  "targets": {
    "test": {
      "executor": "@nx/jest:jest",
      "outputs": [
        "{workspaceRoot}/coverage/{projectRoot}"
      ],
      "options": {
        "jestConfig": "libs/desktop/shared/utils-testing/jest.config.ts",
        "passWithNoTests": true
      },
      "configurations": {
        "ci": {
          "ci": true,
          "codeCoverage": true
        }
      }
    },
    "lint": {
      "executor": "@nx/linter:eslint",
      "outputs": [
        "{options.outputFile}"
      ],
      "options": {
        "lintFilePatterns": [
          "libs/desktop/shared/utils-testing/**/*.ts",
          "libs/desktop/shared/utils-testing/**/*.html"
        ]
      }
    }
  }
}
