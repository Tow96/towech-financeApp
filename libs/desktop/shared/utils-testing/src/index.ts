export * from './lib/shared-data-access-toast.service.mock';
export * from './lib/shared-data-access-user.service.mock';
export * from './lib/shared-utils-toast.stub';
