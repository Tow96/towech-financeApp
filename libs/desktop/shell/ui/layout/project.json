{
  "name": "desktop-shell-ui-layout",
  "$schema": "../../../../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/desktop/shell/ui/layout/src",
  "prefix": "towech-finance",
  "tags": [],
  "projectType": "library",
  "targets": {
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/{projectRoot}"],
      "options": {
        "jestConfig": "libs/desktop/shell/ui/layout/jest.config.ts",
        "passWithNoTests": true
      },
      "configurations": {
        "ci": {
          "ci": true,
          "codeCoverage": true
        }
      }
    },
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": [
          "libs/desktop/shell/ui/layout/**/*.ts",
          "libs/desktop/shell/ui/layout/**/*.html"
        ]
      }
    }
  }
}
