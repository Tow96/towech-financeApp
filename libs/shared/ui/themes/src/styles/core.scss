// The theming functions are heavily taken from angular material, but in a lighter way
// Original code: https://material.angular.io/
@use 'sass:map';

@function get-palette-hue($palette, $hue) {
  @return map.get($palette, $hue);
}

@function get-palette-hue-contrast($palette, $hue) {
  @return map.get(map.get($palette, contrast), $hue);
}

@function define-theme(
  $background,
  $text,
  $primary,
  $accent,
  $success: null,
  $warning: null,
  $error: null
) {
  @if $background == null or $text == null {
    @error 'Background and text colors need to be defined';
  }

  @if $primary == null or $accent == null {
    @error 'primary and accent colors need to be defined palettes';
  }

  @if not map.get($primary, contrast) {
    @error 'Primary is not a palette';
  }

  @if not map.get($accent, contrast) {
    @error 'Accent is not a palette';
  }

  @if $success != null and not map.get($success, contrast) {
    @error 'Success is not a palette';
  }

  @if $error != null and not map.get($error, contrast) {
    @error 'Error is not a palette';
  }

  @if $warning != null and not map.get($warning, contrast) {
    @error 'Warning is not a palette';
  }

  $result: (
    color: (
      text: $text,
      background: $background,
      primary: $primary,
      accent: $accent,
      success: if($success != null, $success, $accent),
      warning: if($warning != null, $warning, $accent),
      error: if($error != null, $error, $accent),
    ),
  );

  @return $result;
}

@mixin color($theme) {
  $colors: map.get($theme, color);
  $primary: map.get($colors, primary);

  :root {
    background-color: map.get($colors, background);
    color: map.get($colors, text);
  }

  div.container {
    background-color: get-palette-hue($primary, 500);
  }
}
