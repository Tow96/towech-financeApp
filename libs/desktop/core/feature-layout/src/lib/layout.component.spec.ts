// Libraries
import { ComponentFixture, TestBed } from '@angular/core/testing';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { subscribeSpyTo } from '@hirez_io/observer-spy';
// Tested elements
import { DesktopLayoutComponent } from './layout.component';
// Services
import { Router } from '@angular/router';
import { DesktopUserService } from '@finance/desktop/shared/data-access-user';
// Mocks
import { DesktopUserServiceMock } from '@finance/desktop/shared/utils-testing';
import { provideStore } from '@ngrx/store';

describe('Desktop Layout', () => {
  let component: DesktopLayoutComponent;
  let fixture: ComponentFixture<DesktopLayoutComponent>;
  let compiled: HTMLElement;
  let router: Router;

  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [BrowserAnimationsModule],
      providers: [
        {
          provide: DesktopUserService,
          useValue: DesktopUserServiceMock,
        },
        provideStore(),
      ],
    });
    router = TestBed.inject(Router);

    fixture = TestBed.createComponent(DesktopLayoutComponent);
    component = fixture.componentInstance;

    fixture.detectChanges();
    compiled = fixture.nativeElement;
  });

  it('Should be defined', () => expect(component).toBeTruthy());

  it('Must match the snapshot', () => expect(compiled).toMatchSnapshot());

  describe('When the router is triggered', () => {
    it('Should cycle the loading spinner', () => {
      const spy = subscribeSpyTo(component.isRouterLoading$);
      router.navigate(['test']);
      expect(spy.getLastValue()).toBeTruthy();
    });
  });
});
