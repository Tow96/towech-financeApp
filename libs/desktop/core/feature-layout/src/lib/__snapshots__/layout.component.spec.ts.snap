// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Desktop Layout Must match the snapshot 1`] = `
<div
  id="root1"
>
  <div
    class="layout"
  >
    <finance-navbar>
      <div
        class="nav-wrapper"
      >
        <nav
          class="deployed"
        >
          <div
            class="header"
          >
            <finance-navbar-item
              icon="bars"
              id="navbar-menu"
              label="Close"
            >
              <button>
                <div>
                  <fa-icon
                    class="ng-fa-icon"
                    size="2x"
                  >
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-bars fa-fw fa-2x"
                      data-icon="bars"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      viewBox="0 0 448 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M0 96C0 78.3 14.3 64 32 64H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H32C14.3 128 0 113.7 0 96zM0 256c0-17.7 14.3-32 32-32H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H32c-17.7 0-32-14.3-32-32zM448 416c0 17.7-14.3 32-32 32H32c-17.7 0-32-14.3-32-32s14.3-32 32-32H416c17.7 0 32 14.3 32 32z"
                        fill="currentColor"
                      />
                    </svg>
                  </fa-icon>
                  <div
                    class="label ng-trigger ng-trigger-pesto"
                  >
                     Close 
                  </div>
                  
                </div>
              </button>
            </finance-navbar-item>
            <div
              class="title"
            >
              <h1>
                
              </h1>
            </div>
          </div>
          <div
            class="contents"
          >
            <div
              class="nav__divider desk__only"
            />
            <div
              class="contents__main"
            >
              
            </div>
            <div
              class="nav__divider"
            />
            <div>
              <finance-navbar-item
                id="logout"
                label="Logout"
              >
                <button>
                  <div>
                    <fa-icon
                      class="ng-fa-icon"
                      size="2x"
                    >
                      <svg
                        aria-hidden="true"
                        class="svg-inline--fa fa-right-from-bracket fa-fw fa-2x"
                        data-icon="right-from-bracket"
                        data-prefix="fas"
                        focusable="false"
                        role="img"
                        viewBox="0 0 512 512"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M377.9 105.9L500.7 228.7c7.2 7.2 11.3 17.1 11.3 27.3s-4.1 20.1-11.3 27.3L377.9 406.1c-6.4 6.4-15 9.9-24 9.9c-18.7 0-33.9-15.2-33.9-33.9l0-62.1-128 0c-17.7 0-32-14.3-32-32l0-64c0-17.7 14.3-32 32-32l128 0 0-62.1c0-18.7 15.2-33.9 33.9-33.9c9 0 17.6 3.6 24 9.9zM160 96L96 96c-17.7 0-32 14.3-32 32l0 256c0 17.7 14.3 32 32 32l64 0c17.7 0 32 14.3 32 32s-14.3 32-32 32l-64 0c-53 0-96-43-96-96L0 128C0 75 43 32 96 32l64 0c17.7 0 32 14.3 32 32s-14.3 32-32 32z"
                          fill="currentColor"
                        />
                      </svg>
                    </fa-icon>
                    <div
                      class="label ng-trigger ng-trigger-pesto"
                    >
                       Logout 
                    </div>
                    
                  </div>
                </button>
              </finance-navbar-item>
            </div>
          </div>
        </nav>
      </div>
    </finance-navbar>
    <div
      class="layout__contents"
    >
      
      <router-outlet />
      
    </div>
  </div>
</div>
`;
